proto-lens-protoc/src/Data/ProtoLens/Compiler/Plugin.hs

```diff
@@ -0,0 +1,125 @@
+-- Copyright 2016 Google Inc. All Rights Reserved.
+--
+-- Use of this source code is governed by a BSD-style
+-- license that can be found in the LICENSE file or at
+-- https://developers.google.com/open-source/licenses/bsd
+--
+-- Code for writing protocol compiler plugins.
+
+{-# LANGUAGE OverloadedStrings #-}
+module Data.ProtoLens.Compiler.Plugin
+    ( ProtoFileName
+    , ProtoFile(..)
+    , analyzeProtoFiles
+    , buildEnv
+    , outputFilePath
+    , moduleName
+    , moduleNameStr
+    ) where
+
+import Data.Char (toUpper)
+import Data.List (foldl', intercalate)
+import qualified Data.Map.Strict as Map
+import Data.Map.Strict (Map, unions, (!))
+import Data.Monoid ((<>))
+import qualified Data.Text as T
+import Data.Text (Text)
+import Language.Haskell.Exts.Syntax (ModuleName(..), Name(..), QName(..))
+import Lens.Family2
+import Bootstrap.Proto.Google.Protobuf.Descriptor
+    (FileDescriptorProto, name, dependency, publicDependency)
+import System.FilePath (dropExtension, splitDirectories)
+
+
+import Data.ProtoLens.Compiler.Definitions
+
+-- | The filename of an input .proto file.
+type ProtoFileName = Text
+
+data ProtoFile = ProtoFile
+  { descriptor :: FileDescriptorProto
+  , haskellModule :: ModuleName
+  , definitions :: Env Name
+  , exports :: [ProtoFileName]
```

Comment:
For `exports`, maybe copy some of the comments at l.104 or else just point to that function's haddocks?  (It took a minute to remind myself how this field worked with everything else.)